pkg/component/status/status.go

```diff
@@ -67,8 +64,7 @@ func (s *Status) Init(_ context.Context) error {
 		Handler: mux,
 	}
 
-	removeLeftovers(s.Socket)
-	s.listener, err = net.Listen("unix", s.Socket)
+	s.listener, err = newStatusListener(s.Socket)
```

Comment:
I know this has been here before, but I'd say this would be better located in the `Start` method, as `Init` is not supposed to open resources that need a cleanup (i.e. if `Init` fails, there's no obligation to call `Stop`). As a plus, we'd get rid of the listener field in the `Status` struct.